serde_derive/src/de.rs

```diff
@@ -1384,7 +1399,36 @@ fn deserialize_identifier(
         "field identifier"
     };
 
-    let visit_index = if is_variant {
+
+    let (bytes_to_str, int_to_str, bool_to_str) = if fallthrough.is_some() {
+        (None, None, None)
+    } else {
+        let conversion_bytes = quote! {
+            let __value = &_serde::export::from_utf8_lossy(__value);
+        };
+        let conversion_int = quote! {
+            let __value = &_serde::export::from_int(__value);
+            let __value = &_serde::export::from_utf8_lossy(__value);
+        };
+        let conversion_bool = quote! {
+            let __value = _serde::export::from_bool(__value);
+        };
+        (Some(conversion_bytes), Some(conversion_int), Some(conversion_bool))
+    };
+
+    let fallthrough_arm = if let Some(fallthrough) = fallthrough {
+        fallthrough
+    } else if is_variant {
+        quote! {
+            _serde::export::Err(_serde::de::Error::unknown_variant(__value, VARIANTS))
+        }
+    } else {
+        quote! {
+            _serde::export::Err(_serde::de::Error::unknown_field(__value, FIELDS))
+        }
+    };
+
+    let visit_int = if constructors_ints.is_empty() && is_variant {
```

Comment:
Note when a deserialized int is interpreted as index and when as a `renamed=int`